src/services/Api.ts

```diff
@@ -47,7 +47,7 @@ const basicAuth = (username: string, password: string) => {
 };
 
 const newRequest = <T>(method: HTTP_VERBS, url: string, queryParams: any, data: any, auth?: AuthToken) => {
-  return new Promise<Response<Readonly<T>>>((resolve, reject) => {
+  return new Promise<Response<T>>((resolve, reject) => {
```

Comment:
@mtho11 I had to remove `Readonly` here (I think you introduced it?), because I call a DELETE endpoint that doesn't return json but just a string, which is incompatible with Readonly (ts error)

Reply:
BTW ... it seems weird here that the axios promise is embedded into a new promise, which doesn't add any extra thing. Or was it rpecisely because of this `Readonly` ? It seems that the same could be done with just:

```
const newRequest = <T>(method: HTTP_VERBS, url: string, queryParams: any, data: any, auth?: AuthToken): Promise<Response<Readonly<T>>> => {
  return axios({
    method: method,
    url: url,
    data: data,
    headers: getHeaders(auth),
    params: queryParams
  });
};
```